Hypothesis Test for a Population Proportion

$$Z_{score} = \frac{\hat{p} - p}{\sqrt{\frac{p*q}{n}}}$$

where:

+ p is the population proportion

+ q = 1 - p

+ $\hat{p}$ is the sample proportion

+ n is the sample size

Exercise 1: For a certain genetic cross it is predicted that one quarter of the offspring will be deficient in a particular enzyme.

Of 48 offspring examined, 8 exhibited this deficiency. Test whether these data are consistent with the theoretical proportion.


$$H_o: p = 1/4$$


$$H_1: p != 1/4$$

In [19]:
n = 48
p = 1/4
q = 1 - p
p_hat = 8/48

# Compute test statistic Z_score
import math
Z_score = (p_hat - p)/math.sqrt((p*q)/n)
print("The test statistic Z_score =", Z_score)

# Compute p_value
from scipy.stats import norm
# We use standard normal distribution (mu =0 and sigma = 1), thus loc = 0 and scale = 1
p_value = norm(loc = 0, scale = 1).cdf(Z_score)
print("p_value =", p_value)


The test statistic Z_score = -1.3333333333333335
p_value = 0.09121121972586782


Since this test is a 2-tail test,

We compare p_value vs $\alpha$/2

And our p_value > $\alpha$/2 (0.091 > 0.05/2)

We do reject $H_o$

Conclusion: The data are consistent with the null hypothesis which is the proportion of the enzyme deficiency is 25%. 


Exercise 2: A cell culture method usually has a failure rate of 20% (caused by aseptic technique failing etc.). Using a changed method, six failures occurred in 50 test cultures. Does this imply that the change has reduced the failure rate?

$$H_o: p = 0.2$$


In [20]:
n = 50
p = 0.2
q = 1 - p
p_hat = 6/50

# Compute test statistic Z_score
import math
Z_score = (p_hat - p)/math.sqrt((p*q)/n)
print("The test statistic Z_score =", Z_score)

# Compute p_value
from scipy.stats import norm
# We use standard normal distribution (mu =0 and sigma = 1), thus loc = 0 and scale = 1
"""If Z_score > 0, p_value = 1 - norm(loc = 0, scale = 1).cdf(Z_score)
   If Z_score < 0, p_value = norm(loc = 0, scale = 1).cdf(Z_score)"""

if Z_score > 0:
    p_value = 1 - norm(loc = 0, scale = 1).cdf(Z_score)
    print("p_value =", p_value)
else:
    p_value = norm(loc = 0, scale = 1).cdf(Z_score)
    print("p_value =", p_value)


The test statistic Z_score = -1.4142135623730951
p_value = 0.07864960352514251


Since p_value > alpha (0.079 > 0.05),

We do not reject $H_o$

Conclusion: The changed method still has a failure rate of 20%